scrapper.py

```diff
@@ -25,39 +39,136 @@ class Crawler:
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        """
+        Extracts urls of articles
+        """
+        articles = article_bs.find_all('div', class_='block-container')
+
+        for article in articles:
+            if len(self.urls) < self.max_articles:
+                link = article.find('a')
+                href = link['href']
+                self.urls.append(href)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            sleep(randint(1, 5))
+            response = requests.get(seed_url, headers=HEADERS, allow_redirects=True)
+
+            article_bs = BeautifulSoup(response.text, 'lxml')
+            self.urls.extend(self._extract_url(article_bs))
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.urls
 
 
-def prepare_environment(base_path):
+class HTMLParser:
     """
-    Creates ASSETS_PATH folder if not created and removes existing folder
+    Parser implementation
     """
-    pass
 
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
 
-def validate_config(crawler_path):
-    """
-    Validates given config
-    """
-    pass
+    def parse(self):
+        """
+        Extracts all necessary data from the article web page
+        """
+        response = requests.get(self.article_url, headers=HEADERS, allow_redirects=True)
+
+        article_bs = BeautifulSoup(response.text, 'lxml')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills the Article instance with text
+        """
+        text_bs = article_bs.find(class_="text-article").text
+        self.article.text = text_bs.text
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills the Article instance with meta information
+        """
+        title_bs = article_bs.find(class_='top_big_img_article__info__inside__title').text
+        self.article.title = title_bs
+        print(title_bs)
+
+        date_bs = article_bs.find(class_="top_big_img_article__info__inside__time").text
+        date = datetime.strptime(date_bs, "%d.%m.%Y, %H:%M")
+        self.article.date = date
+
+        tags_bs = article_bs.find('div', class_='hash_tags')
+        self.article.tags = tags_bs
+
+        try:
+            author_bs = article_bs.find(class_='top_big_img_article__info__inside__author').text
+        except AttributeError:
+            author_bs = 'NOT FOUND'
+        self.article.author = author_bs
+
+    def prepare_environment(base_path):
```

Comment:
It is not a responsibility of Parser to prepare environment for crawling
There is no need to turn it into its method